### DISCUSSION TITLE
Math doesn't render

### DISCUSSION THREAD

**Discussion by achimstruve** [facebook/docusaurus#9228] [Q&A] [answered]
I can't get math equations getting rendered by using KaTex in my markdown files and don't know why. Can anyone help me, please?

I created a new clean [docusaurus project](https://github.com/achimstruve/docusaurus_math_minimal) and afterwards followed the exact [instructions on how to include KaTex math](https://docusaurus.io/docs/markdown-features/math-equations) within that project.

In this clean project I created an example page under `./src/pages/markdown-math-page.mdx`.
```---
title: Markdown page example
---

# Markdown page math example

Test math:
$a
$a + b = c$
```

It renders as shown in the image below, but the math equations aren't rendered as they should.
What step am I missing?
![image](https://github.com/facebook/docusaurus/assets/25647387/607f38d5-4e92-45b1-8855-0b95254766e9)

**Comment by slorber** [comment]
Please invest some time to create a repro

**Reply by achimstruve** [reply]
I created a minimal example repo just for this question as stated in the initial question. Did you see it? Would really appreciate any help, thanks!

**Reply by slorber** [reply]
Sorry missed it.

**Comment by slorber** [comment, marked as answer]
Repro: https://stackblitz.com/~/github.com/achimstruve/docusaurus_math_minimal

Your math equations works, but only where you configured them: on the docs

<img width="1505" alt="CleanShot 2023-08-17 at 12 06 12@2x" src="https://github.com/facebook/docusaurus/assets/749374/b4a9ed62-3fe8-4406-af48-b440e2454eff">


If you want maths to work on pages, you have to add the remark/rehype plugins to your pages plugin as well (and not just the docs plugin). 

I understand how it can be confusing to have to register the same remark/rehype plugins multiple times, this will be solved in https://github.com/facebook/docusaurus/issues/6370

**Reply by achimstruve** [reply]
Thank you very much! Your answer helped me solving it!

I just changed the presets to import the remark/rehype plugins for the `src/pages` path as follows. This is my new `docusaurus.config.js`:

```const math = require("remark-math");
const katex = require("rehype-katex");

module.exports = {
  title: "Docusaurus",
  tagline: "Build optimized websites quickly, focus on your content",

  url: "https://your-docusaurus-test-site.com",
  baseUrl: "/",

  presets: [
    [
      "@docusaurus/preset-classic",
      {
        docs: {
          path: "docs",
          remarkPlugins: [math],
          rehypePlugins: [katex],
        },
        pages: {
          path: "src/pages",
          remarkPlugins: [math],
          rehypePlugins: [katex],
        },
      },
    ],
  ],
  stylesheets: [
    {
      href: "https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css",
      type: "text/css",
      integrity:
        "sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM",
      crossorigin: "anonymous",
    },
  ],
};```
